ext/standard/uuid.c

```
@@ -0,0 +1,646 @@
+	php_uuid uuid;
+
+	if (zend_parse_parameters_ex(ZEND_PARSE_PARAMS_QUIET, MIN(ZEND_NUM_ARGS(), 1), "s", &input, &input_len) == FAILURE) {
+		throw_argument_count_error("parse", ZEND_NUM_ARGS(), 1);
+		return;
+	}
+
+	php_uuid_parse_throw(&uuid, input, input_len);
+	new_uuid(return_value, uuid);
+}
+
+PHP_METHOD(UUID, v3)
+{
+	zval *namespace = NULL;
+	char *name      = NULL;
+	size_t name_len;
+	php_uuid uuid;
+
+	if (zend_parse_parameters_ex(ZEND_PARSE_PARAMS_QUIET, MIN(ZEND_NUM_ARGS(), 2), "zs", &namespace, &name, &name_len) == FAILURE) {
+		throw_argument_count_error("v3", ZEND_NUM_ARGS(), 2);
+		return;
+	}
+
+	php_uuid_create_v3(&uuid, get_bytes(namespace), name, name_len);
+	new_uuid(return_value, uuid);
+}
+
+PHP_METHOD(UUID, v4)
+{
+	php_uuid uuid;
+	if (php_uuid_create_v4_throw(&uuid) == SUCCESS) {
+		new_uuid(return_value, uuid);
+	}
+}
+
+PHP_METHOD(UUID, v5)
+{
+	zval *namespace = NULL;
+	char *name      = NULL;
+	size_t name_len;
+	php_uuid uuid;
+
+	if (zend_parse_parameters_ex(ZEND_PARSE_PARAMS_QUIET, MIN(ZEND_NUM_ARGS(), 2), "zs", &namespace, &name, &name_len) == FAILURE) {
+		throw_argument_count_error("v5", ZEND_NUM_ARGS(), 2);
+		return;
+	}
+
+	php_uuid_create_v5(&uuid, get_bytes(namespace), name, name_len);
+	new_uuid(return_value, uuid);
+}
+
+PHP_METHOD(UUID, NamespaceDNS)  { new_uuid(return_value, PHP_UUID_NAMESPACE_DNS);  }
+PHP_METHOD(UUID, NamespaceOID)  { new_uuid(return_value, PHP_UUID_NAMESPACE_OID);  }
+PHP_METHOD(UUID, NamespaceURL)  { new_uuid(return_value, PHP_UUID_NAMESPACE_URL);  }
+PHP_METHOD(UUID, NamespaceX500) { new_uuid(return_value, PHP_UUID_NAMESPACE_X500); }
+PHP_METHOD(UUID, Nil)           { new_uuid(return_value, PHP_UUID_NIL);            }
+
+PHP_METHOD(UUID, __clone) { zend_throw_error(zend_ce_error, "Cannot clone immutable UUID object");                     }
+PHP_METHOD(UUID, __set)   { zend_throw_error(zend_ce_error, "Cannot set dynamic properties on immutable UUID object"); }
+
+PHP_METHOD(UUID, __wakeup)
+{
```

If it weren't for PHP function signatures being ambiguous with respect to the number of required arguments (due to implicit variadics with `func_get_args()`), passing more than zero arguments to a userland function with zero parameters would produce an error. We already throw an error for passing too few arguments. And indeed, all internal functions with zero expected arguments throw such errors if you pass any.

I don't understand the compatibility argument. Why should you allow passing arguments now that might drastically alter the function's behaviour in future? You can't possibly know what they will do right now, and if they're added in future, having them silently ignored (and therefore meaning your call does something different) in old versions is unhelpful.